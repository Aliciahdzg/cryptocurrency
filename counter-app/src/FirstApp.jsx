
import * as React from 'react';
import PropTypes from 'prop-types';

const FirstApp = ({ title, subtitle, name }) => {
    
  return (
    <>
      <h1 data-testid="test-title">{ title }</h1>
      <p>{ subtitle }</p>
      <p>{ name }</p>
    </>
  )
}

export default FirstApp

FirstApp.propTypes = { 
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired, 
    name: PropTypes.string.isRequired
}

// FirstApp.defaultProps = {
//     // title: 'No title',
//     subtitle: 'No subtitle',
//     name: 'Alicia Hernandez'
// }