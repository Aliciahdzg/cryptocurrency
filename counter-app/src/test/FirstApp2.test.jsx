import { render, screen } from "@testing-library/react"
import FirstApp from '../FirstApp';


describe('Pruebas en  <FirstApp />', () => {
    const name = 'Alicia Hernandez';
    const title = 'My FirstApp';
    const subtitle = 'Hola Soy:';

    test('debe hacer match con el snapshot', () => {
        const {container} = render( <FirstApp name={name} title={title} subtitle={subtitle} />);

        expect( container ).toMatchSnapshot();

    });

    test('debe mostrar el mensaje My FirstApp', () => {
         
        render( <FirstApp name={name} title={title} subtitle={subtitle} /> );
        expect( screen.getByText(title) ).toBeTruthy();
    });

    test('debe mostrar el titulo en un h1', () => {

        render( <FirstApp name={name} title={title} subtitle={subtitle} /> );
        expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain(title);
    });

    test('debe mostrar el subtitulo enviado por props', () => {
        render( 
          <FirstApp 
            name={name} 
            title={title} 
            subtitle={subtitle} 
          /> 
        );
        expect( screen.getByText( subtitle ) ).toBeTruthy();
    })
})