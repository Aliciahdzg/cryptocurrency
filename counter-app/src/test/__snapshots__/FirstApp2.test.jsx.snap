// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en  <FirstApp /> debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    My FirstApp
  </h1>
  <p>
    Hola Soy:
  </p>
  <p>
    Alicia Hernandez
  </p>
</div>
`;
